components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/SAML2SSOManager.java

```diff
@@ -365,6 +368,81 @@ public void processResponse(HttpServletRequest request, HttpServletResponse resp
         }
     }
 
+    /**
+     * Process authentication response with SAML2 artifact.
+     *
+     * @param request Http Servlet Request object.
+     * @throws SSOAgentException
+     */
+    public void processArtifactResponse(HttpServletRequest request) throws SSOAgentException {
+
+        SAMLSSOArtifactResolutionService artifactResolutionService =
+                new SAMLSSOArtifactResolutionService(ssoAgentConfig);
+        try {
+            ArtifactResponse artifactResponse = artifactResolutionService.getSAMLArtifactResponse(
```

Comment:
Would be better to add debug log to print artifact response.

Reply:
Printing artifact response in [here](https://github.com/wso2-extensions/identity-agent-sso/pull/46/files#diff-e58b1bc8dd5cd23b7eb0aa3b3e5d0a68R175).